pybossa/view/account.py

```diff
@@ -1064,6 +1065,38 @@ def add_metadata(name):
     flash("Input saved successfully", "info")
     return redirect(url_for('account.profile', name=name))
 
+def get_form_data(request, user, can_update):
+    if isinstance(can_update, dict):
```

Comment:
In this function we do not handle `can_update is False`. I will put a comment to that effect. We don't handle it because the caller has already returned a 403 in that case.

So to do your suggestion I would change the parameter `can_update` of type `bool | dict` to `disabled_fields` of type `dict`. If nothing is disabled then it would be empty or `None`, both of which are Falsy. So it would be something like:
```
# This is only called if can_update is True.
def get_form_data(request, user, disabled_fields):
    if not disabled_fields:
        return request.form
    # Handle disabled fields
```